spec/features/course/assessment/submission/submissions_spec.rb

```diff
@@ -42,6 +49,29 @@
         create(:course_group_student, course: course, group: group, course_user: students.sample)
       end
 
+      scenario 'I can directly publish prefilled grade', js: true do
+        # attempting the question in which grade is prefillable upon grading
+        visit edit_course_assessment_submission_path(course, mcq_assessment, staff_attempt_submission)
+        find('label', text: correct_option.option).click
+
+        click_button('Finalise Submission')
+        click_button('Continue')
+        wait_for_page
+
+        visit edit_course_assessment_submission_path(course, mcq_assessment, staff_attempt_submission)
+
+        click_button('Publish Grade')
+        wait_for_page
+
+        expect(page).to have_selector('span', text: 'Submission updated successfully.')
+
+        # since the answer is correct, the submission's question should not have zero grade
+        staff_attempt_submission.answers.each do |answer|
+          expect(answer.reload.grade).not_to eq(0)
```

Comment:
check for the actual exact grade.

Reply:
yes, the line after this check for that actual exact grade. Basically if the grade is prefilled, it will always be the maximum grade for the corresponding question

Reply:
Then why do we need to check if it’s not zero?

Reply:
I'm sorry, you're right. It's an oversight from my side that the maximum grade can actually be 0. I have deleted this line